frontend/src/metabase/search/components/SearchUserPicker/SearchUserPicker.tsx

```diff
@@ -49,19 +62,57 @@ export const SearchUserPicker = ({
   return (
     <SearchFilterPopoverWrapper
       isLoading={isLoading}
-      onApply={() => onChange(selectedUserId)}
+      onApply={() => onChange(selectedUserIds)}
     >
-      <SearchUserPickerContainer p="sm" h="100%" spacing="xs">
-        <TextInput
-          size="md"
-          mb="sm"
-          placeholder={t`Search for users…`}
-          value={userFilter}
-          tabIndex={0}
-          onChange={event => setUserFilter(event.currentTarget.value)}
-        />
+      <SearchUserPickerContainer p="sm">
+        <SearchUserSelectBox spacing={0}>
```

Comment:
The JSX here adds the logic for the text input and user chips, and ensures that the area containing chips doesn't overflow into the list of users and off the screen.